### ISSUE TITLE
Prefetch_related field name gets continuously appended when type is used as a related field

### ISSUE THREAD

**Issue by hlschmidbauer** [strawberry-graphql/strawberry#3176]
Subsequent queries fail with error `invalid parameter to prefetch_related()` when attempting to query for a field `bar` on a type `foo` where `foo` is a related field and `bar` defines a prefetch via `@strawberry_django.field(prefetch_related=Prefetch(...))`

If the `prefetch_related` is removed, this issue does not occur. 

## Describe the Bug

Given
```
@strawberry.django.type(Baz)
class BazType:
    foo: FooType

@strawberry.django.type(Foo)
class FooType:
    @strawberry_django.field(
        prefetch_related=Prefetch(
            "bars",
             queryset=Bar.objects.active().select_related("qux")
            to_attr="_bar_prefetch",
        )
    )
    def bar(self) -> Optional[
        Annotated["BarType", strawberry.lazy("..qux")]
    ]: 
        return self.bar
```

A query for `baz` that attempts to retrieve `bar` on `foo`:
```
query GetBaz {
    results {
        id
        foo {
            bar {
                id
            }
        }
```

Succeeds on the first execution, but on the second execution fails with 
`"Cannot find 'foo' on Foo object, 'foo_foo_bars' is an invalid parameter to prefetch_related()"`, then on the third execution fails with `"Cannot find 'foo' on Foo object, 'foo_foo_foo_bars' is an invalid parameter to prefetch_related()"`, on the fourth with `"Cannot find 'foo' on Foo object, 'foo_foo_foo_foo_bars' is an invalid parameter to prefetch_related()"` and so on...

However, once the `prefetch_related` is removed from the `bar` field definition, this issue does not occur. 

## System Information

 - Operating system: MacOS/Linux
 - Python Version: `3.9`
 - Strawberry version (if applicable): `==0.209.1`

## Additional Context

We thought that this issue was fixed with the latest library update, but it has persisted.

**Comment by patrick91** [comment]
@bellini666 this is for you 😊

**Comment by bellini666** [comment]
@hlschmidbauer what version of strawberry_django are you using? This issue was resolved on 0.19.0

Also, one recommendation is to use a lambda there, like: `prefetch_related=lambda info: Prefetch(...)`, that way the prefetch object will for sure be unique all the time.

That was actually the original issue that was fixed in 0.19.0 by making a copy of it when adding a prefix on it.

**Comment by hlschmidbauer** [comment]
> @hlschmidbauer what version of strawberry_django are you using? This issue was resolved on 0.19.0
> 
> Also, one recommendation is to use a lambda there, like: `prefetch_related=lambda info: Prefetch(...)`, that way the prefetch object will for sure be unique all the time.
> 
> That was actually the original issue that was fixed in 0.19.0 by making a copy of it when adding a prefix on it.

@bellini666 Ah thank you! We are still on `strawberry-graphql-django = "==0.17.3"`, so that would definitely explain it. Apologies for the churn here!
